### PR TITLE
Performance

### PR DESCRIPTION
**Pull request by jbrundage**
Adds timestamps in Widget preUpdate and postUpdate to keep track of update times.

http://rawgit.com/jbrundage/Visualization/Performance/demos/dermatology.html?src/composite/Performance.column
http://rawgit.com/jbrundage/Visualization/Performance/demos/dermatology.html?src/composite/Performance.bar
http://rawgit.com/jbrundage/Visualization/Performance/demos/dermatology.html?src/composite/Performance.line
http://rawgit.com/jbrundage/Visualization/Performance/demos/dermatology.html?src/composite/Performance.area
http://rawgit.com/jbrundage/Visualization/Performance/demos/dermatology.html?src/composite/Performance.pie

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage What does this give you over the built in profiling tools in Chrome +IE?  Also have you looked at console.profile?

**Comment by jbrundage** [comment]
The browser profiling tools are a more accurate test... this just visually compares our different chart libraries

**Comment by jbrundage** [comment]
I havent used console.profile, I'll try it out

File: src/other/PropertyEditor.js [hunk 1]
```diff
@@ -129,11 +129,14 @@
         }
         if (widget) {
             var context = this;
-            this._watch = widget.monitor(function (paramId, newVal, oldVal) {
-                if (oldVal !== newVal) {
-                    context.lazyRender();
-                }
-            });
+            if(typeof widget.watch === "function"){
```
Comment (GordonSmith) [inline src/other/PropertyEditor.js]: "watch" has been replaced with "monitor" due to clash with Object.watch on FF - I suspect you may need to rebase your code?

File: src/common/Widget.js [hunk 2]
```diff
@@ -483,10 +483,39 @@
...
+                    diff.push(context._postUpdateTime[i] - context._preUpdateTime[i]);
+                });
+            }
+        }
+        var sum = 0;
+        diff.forEach(function(n){
+            sum += n;
+        });
+        return sum/diff.length;
+    };
```
Comment (GordonSmith) [inline src/common/Widget.js]: Can you put the contents of each method into an ```__hpcc_debug``` test?

**Comment by GordonSmith** [comment]
@jbrundage bump

**Comment by jbrundage** [comment]
@GordonSmith Should be good now

**Comment by jbrundage** [comment]
I'm working on resolving this at the moment...

**Comment by jbrundage** [comment]
It's failing the unit testing because it's attempting to load the line/area/column/bar/pie charts with 16px height.

Maybe it would be best to close this pull request? It was more of an experiment than a feature.

**Pull request closed without merging**